docs/source/isolation.rst

```diff
@@ -31,64 +29,55 @@ How Tasks are Scheduled on Hosts
 --------------------------------
 
 To first understand how these resources are used, one must understand how
-a task is run on a Mesos cluster.
-
-Mesos can run in two modes: Master and Agent. When a node is running Mesos in
-Master mode, it is responsible for communicating between agent processes and
-frameworks. A Framework is a program which wants to run tasks on the Mesos
-cluster.
-
-A master is responsible for presenting frameworks with resource offers.
-Resource offers are compute resource free for a framework to run a task. The
-details of that compute resource comes from the agent nodes, which regularly
-tell the Master agent the resources it has available for running tasks. Using
-the correct parlance, Mesos agents make 'offers' to the master.
-
-Once a master node receives offers from an agent, it forwards it to
-a framework. Resource offers are split between frameworks according to
-the master's configuration - there may be particular priority given
-to some frameworks.
-
-At Yelp, we treat the frameworks we run (at the time of writing, Marathon and
-Tron) equally. That means that frameworks *should* have offers distributed
-between them evenly, and all tasks are considered equal.
-
-It is then up to the framework to decide what it wants to do with an offer.
-The framework may decide to:
-
-  * Reject the offer, if the framework has no tasks to run.
-  * Reject the offer, if the resources included in the offer are not enough to
-    match those required by the application.
-  * Reject the offer, if attributes on the slave conflict with any constraints
-    set by the task.
-  * Accept the offer, if there is a task that requires resources less than or
-    equal to the resources offered by the Agent.
-
-When rejecting an offer, the framework may apply a 'filter' to the offer. This
-filter is then used by the Mesos master to ensure that it does *not* resend
-offers that are 'filtered' by a framework. The default filter applied includes
-a timeout - a Master will not resend an offer to a framework for a period of 5
-seconds.
-
-If a framework decides it wants to accept a resource offer, it then tells the
-master to run a task on the agent. The details of the 'acceptance' include a
-detail of the task to be run, and the 'executor' used to run the task.
-
-By default, PaaSTA uses the 'Docker' executor everywhere. This means that *all*
-tasks launched by Marathon and Tron are done so with a Docker container.
-
-How Tasks are isolated from each other.
+a pod is run on a Kubernetes cluster.
+
+Kubernetes has two types of nodes: Master and worker nodes. The master node is
+is responsible for managing the cluster. This includes scheduling applications (i.e. paasta services), maintaining
+applications' desired states, scaling applications and rolling out new updates.
```

Comment:
this is maybe also a little confusing since we have the usual k8s masters + the paasta system pool - so depending on what type of `maintaining
applications' desired states` or `rolling out new updates` we're talking about, that can happen on either type of box (e.g., the paasta boxes will manage the state that k8s should have and the k8s boxes will manage the actual k8s cluster with that desired state)

Reply:
hmm, maybe we should link to the official k8s docs about how scheduling and whatnot work?

Reply:
yea, I was talking here only from k8s pov and how in general kubernetes master nodes responsibilities are